src/CIQS-CLI-Extension/azext_ciqs/custom.py

```diff
@@ -33,10 +33,31 @@ def listDeployments(cmd, subscription=None):
     profile = Profile(cli_ctx=cmd.cli_ctx)
     auth_token = profile.get_raw_token(subscription=subscription)
     path = api.DEPLOYMENT_ENDPOINT + subscription
-    return api.makeAPICall(cmd, 'GET', path, auth_token=auth_token)
+    return api.makeAPICall('GET', path, auth_token=auth_token)
 
-def createDeployment(cmd, deploymentObj, subscription=None):
-    raise CLIError('Not Implemented')
+def createDeployment(cmd, name, location, templateId, description=None, parameters=None, solutionStorageConnectionString=None, subscription=None):
+    if subscription is None:
+        subscription = get_subscription_id(cmd.cli_ctx)
+    profile = Profile(cli_ctx=cmd.cli_ctx)
+    auth_token = profile.get_raw_token(subscription=subscription)
+    path = api.DEPLOYMENT_ENDPOINT
+    createData = {}
```

Comment:
Can this be modeled as a [class](https://docs.python.org/3/tutorial/classes.html) such that it can be extended and customized:

class CreateDeploymentRequest:
     def __init__(self, name, location, ...):
         self.name = name
         self.location = location
        ...

Reply:
done